src/Dialogs/HelloExampleDialog.php

```diff
@@ -19,6 +19,9 @@ final class HelloExampleDialog extends Dialog
 
     public function sayHello(Update $update): void
     {
+        //Used for test purposes
+        if(!isset($this->bot)){return;}
```

Comment:
this HelloExampleDialog should be sort of showcase for developers, and we should avoid adding such code needed for testing purposes.

if you want to test Dialog-related functionality, please consider either:
 - create a new one that can even not call any
 - use the same approach as a used to create a fake bot that doesn't use HTTP (see https://github.com/koot-labs/telegram-bot-dialogs/blob/master/tests/Fakes/FakeBot.php)

Reply:
Hello, @alies-dev. Thank you for your feedback. So should i cancel this PR and submit another one with fixes?
According your last proposal "use the same approach" - i tried, but failed. Maybe it was because i planned test with sendMessage and did not understand how to emulate its call in fake bot. But as for my last test i do not need this call, so i will try again.

Reply:
@dimaodnokoz 
pls update code for the existing PR, I think it will be simpler to you.

Also, if you have issues with Faking HTTP requests and responses, I think it will be simpler for your to create a new Dialog implementation like `PassiveTestDialog extends Dialog` that only listens and doesn't send anything and use it for your tests.

You can also implement `afterLastStep` in that Dialog  and e.g. throw an exception. If exception is not thrown - the test should fail. This way you will avoid adding code like:
```php
public bool $afterLastStepCalled = false;

    protected function afterLastStep(Update $update): void
    {
        $this->afterLastStepCalled = true;
    }
```
(it's just an idea, you can find another implementation, incl. public properties that are ok for a TestDialog)